orchagent/neighorch.cpp

```diff
@@ -354,13 +354,31 @@ void NeighOrch::doTask(Consumer &consumer)
             if (m_syncdNeighbors.find(neighbor_entry) == m_syncdNeighbors.end() || m_syncdNeighbors[neighbor_entry] != mac_address)
             {
                 if (addNeighbor(neighbor_entry, mac_address))
+                {
                     it = consumer.m_toSync.erase(it);
+                }
                 else
+                {
                     it++;
+                    continue;
+                }
             }
             else
+            {
                 /* Duplicate entry */
                 it = consumer.m_toSync.erase(it);
+            }
+
+            /* Remove remaining DEL operation in m_toSync for the same neighbor.
+             * Since DEL operation is supposed to be executed before SET for the same neighbor [ref: https://github.com/Azure/sonic-swss/pull/1184]
+             * A remaining DEL after the SET operation means the DEL operation failed previously and should not be executed anymore
+             */
+            auto it_rem = consumer.m_toSync.find(key);
+            if (it_rem != consumer.m_toSync.end() && kfvOp(it_rem->second) == DEL_COMMAND)
+            {
+                it = consumer.m_toSync.erase(it_rem);
+                SWSS_LOG_NOTICE("Removed unfinished neighbor DEL operation for %s after SET operation", key.c_str());
```

Comment:
This can be at INFO level.

Reply:
I thought this is an unexpected scenario and it should be made easily noticeable by users. Also, this would only happen once for each key and should not log a lot of redundant messages.

Reply:
Agree, but this looks like an unactionable log. Anyways, if it is not expected to be high even with large number of neighbor entries, lgtm.